code/game/gamemodes/umbrage/umbrage_ability.dm

```diff
@@ -10,6 +10,17 @@
 	var/in_use = FALSE //For channeled/cast-time abilities
 	var/datum/umbrage/linked_umbrage //Our linked umbrage datum
 
+/datum/action/innate/umbrage/New()
+	START_PROCESSING(SSobj, src)
+	..()
+
+/datum/action/innate/umbrage/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/datum/action/innate/umbrage/process()
+	UpdateButtonIcon() //This is to constantly check psi
```

Comment:
honestly this doesn't need to process itself at all; the umbrage datum itself could update your action buttons when you gain/lose psi instead.